apps/client/src/app/pages/db/fish/fish-hooksets/fish-hooksets.component.ts

```diff
@@ -0,0 +1,43 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { SettingsService } from 'apps/client/src/app/modules/settings/settings.service';
+import { combineLatest } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { FishContextService } from '../../service/fish-context.service';
+
+@Component({
+  selector: 'app-fish-hooksets',
+  templateUrl: './fish-hooksets.component.html',
+  styleUrls: ['./fish-hooksets.component.less', '../../common-db.less'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FishHooksetsComponent {
+  public readonly spotIdFilter$ = this.fishCtx.spotId$.pipe(map((spotId) => spotId ?? -1));
+
+  public readonly loading$ = combineLatest([this.fishCtx.hooksetsByFish$, this.fishCtx.tugsByFish$]).pipe(map(([hook, tug]) => hook.loading || tug.loading));
+
+  public readonly hooksets$ = this.fishCtx.hooksetsByFish$.pipe(
+    map((res) => {
+      if (!res.data) return undefined;
+      return Object.values(res.data.byId)
+        .sort((a, b) => b.occurrences - a.occurrences)
+        .map((entry) => ({
+          hookset: entry.id === 1 ? 4103 : 4179,
+          percent: (100 * entry.occurrences) / res.data.total,
+        }));
+    })
+  );
+
+  public readonly tugs$ = this.fishCtx.tugsByFish$.pipe(
+    map((res) => {
+      if (!res.data) return undefined;
```

Comment:
You can return an empty array here so you don't need to use `|| []` inside the template.
```suggestion
      if (!res.data) return [];
```